components/track/TrackList.vue

```diff
@@ -30,8 +61,15 @@ const { setCurrentSong } = inject(MediaPlaylistInjectionKey)!;
         :key="track.id || 0"
         :track="track"
         show-thumbnail
-        @click="setCurrentSong(track.media?.[0]?.files?.[0]?.url || '')"
-      />
+        @play-track="setCurrentSong(track.media?.[0]?.files?.[0]?.url || '')"
+        @open-options="toggleDropdownForTrack(track.id || 0)"
+      >
+        <DropdownMenu
+          v-if="showDropDownForTrackId === track.id"
+          :items="dropdownMenuItemsForTrack(track)"
+          @close="toggleDropdownForTrack(track.id || 0)"
```

Comment:
Taken the case you have one track twice in your playlist. Would this not open/close it for two items on a list?

Reply:
@SimonSimCity oh yeah, thats true. I'll fix it :)

Reply:
@SimonSimCity Should be good now 👍